lib/spack/env/cc

```diff
@@ -440,6 +441,28 @@ while [ $# -ne 0 ]; do
         continue
     fi
 
+    if [ -n "${SPACK_COMPILER_FLAGS_KEEP}" ] ; then
+        eval "\
+        case '$1' in
+            $SPACK_COMPILER_FLAGS_KEEP)
+                append other_args_list "$1"
+                shift
+                continue
+                ;;
+        esac
+        "
```

Comment:
Is the reason this is all within an `eval` statement because `$SPACK_COMPILER_FLAGS_KEEP` might contain a wildcard? Could we add a comment to that effect?

Reply:
Hmm, I've just tested this out in alpine busybox `sh`, and it seems like parameter expansion occurs *before* wildcard expansion, such that:
```sh
case "$1" in
  $SPACK_COMPILER_FLAGS_KEEP)
    append other_args_list "$1"
    shift
    continue
    ;;
esac
```
should work. Could you confirm that, or is there another reason this is wrapped in an `eval` string?

Reply:
It's not, wildcards work fine, but `|` (pipe characters to allow multiple alternatives) do not.

Reply:
Ok, I'd like to have a comment to that effect in case someone figures out a nicer way to implement this later is all. Thanks for clarifying!